roxctl/scanner/uploaddb/uploaddb_test.go

```diff
@@ -1,18 +1,101 @@
 package uploaddb
 
 import (
+	"bytes"
+	"fmt"
 	"io/fs"
+	"net/http"
+	"net/http/httptest"
+	"os"
 	"testing"
 
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/roxctl/common/environment"
+	"github.com/stackrox/rox/roxctl/common/flags"
+	"github.com/stackrox/rox/roxctl/common/printer"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-func TestScannerUploadDbCommandFail(t *testing.T) {
-	cmdNoFile := scannerUploadDbCommand{filename: "non-existing-filename"}
+func executeUpdateDbCommand(serverURL string) (*bytes.Buffer, *bytes.Buffer, error) {
+	tmpFile, errTempFile := os.CreateTemp("", "*.zip")
+	if errTempFile != nil {
+		return nil, nil, errTempFile
+	}
+	defer utils.IgnoreError(func() error {
+		return os.Remove(tmpFile.Name())
+	})
 
-	actualErr := cmdNoFile.uploadDd()
+	testIO, _, stdOut, stdErr := environment.TestIO()
+	env := environment.NewCLIEnvironment(testIO, printer.DefaultColorPrinter())
 
-	require.Error(t, actualErr)
-	assert.ErrorIs(t, actualErr, fs.ErrNotExist)
+	cmd := Command(env)
+	flags.AddTimeout(cmd)
+	flags.AddConnectionFlags(cmd)
+	flags.AddPassword(cmd)
+
+	cmdArgs := []string{"--insecure-skip-tls-verify", "--insecure", "--endpoint", serverURL, "--password", "test"}
+	cmdArgs = append(cmdArgs, "--scanner-db-file", tmpFile.Name())
+	cmd.SetArgs(cmdArgs)
+
+	cmdExecutionErr := cmd.Execute()
+	if cmdExecutionErr != nil {
+		return stdOut, stdErr, cmdExecutionErr
+	}
+
+	return stdOut, stdErr, nil
+}
+
+func TestScannerUploadDbCommand(t *testing.T) {
+	t.Run("file does not exist", func(t *testing.T) {
+		cmdNoFile := scannerUploadDbCommand{filename: "non-existing-filename"}
+
+		actualErr := cmdNoFile.uploadDd()
+
+		require.Error(t, actualErr)
+		assert.ErrorIs(t, actualErr, fs.ErrNotExist)
+	})
+
+	t.Run("server error", func(t *testing.T) {
+		expectedErrorStr := "test-server-error"
+		server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			rw.WriteHeader(http.StatusInternalServerError)
+			_, _ = rw.Write([]byte(expectedErrorStr))
+		}))
+		defer server.Close()
+
+		_, _, err := executeUpdateDbCommand(server.URL)
+
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), expectedErrorStr)
```

Comment:
Thanks to this test we can see that we put anything server returns into error that's presented to user. Probably not the best user experience.

Reply:
@janisz We are always wrapping errors and bubbling them up. I would not change that behavior for now.

That server response is part of the error is part of the implementation of `common.DoHTTPRequestAndCheck200`. I would leave it for now and eventually change it when we switch to `environment.Environment.HTTPClient()`. Unfortunately, similar behavior is implemented there too, check `RoxctlHTTPClient.DoReqAndVerifyStatusCode`.